src/zulip.rs

```diff
@@ -347,22 +347,49 @@ impl Recipient<'_> {
                 let mut encoded_topic = Vec::new();
                 for ch in topic.bytes() {
                     if !(ALWAYS_SAFE.contains(ch as char)) {
-                        write!(encoded_topic, "%{:02X}", ch).unwrap();
+                        write!(encoded_topic, ".{:02X}", ch).unwrap();
                     } else {
                         encoded_topic.push(ch);
                     }
                 }
-
-                let mut encoded_topic = String::from_utf8(encoded_topic).unwrap();
-                encoded_topic = encoded_topic.replace("%", ".");
```

Comment:
This seems to have changed behavior. AFAICT, previously, if `%` was fed in it would be changed to a `.`. Now it will not be.

Reply:
The behaviour doesn't change, because `%` is encoded to `.26` anyway

Reply:
Oh, right, I missed that it's *not* in the safe list. Okay. Thanks!